packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from "vscode";
+
+const ControlCodes = {
+  CtrlC: "\u0003",
+};
+
+export abstract class BaseTaskTerminal implements vscode.Pseudoterminal {
+  protected writeEmitter = new vscode.EventEmitter<string>();
+  onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+  protected closeEmitter = new vscode.EventEmitter<number>();
+  onDidClose?: vscode.Event<number> = this.closeEmitter.event;
+
+  constructor(private taskDefinition: vscode.TaskDefinition) {}
+
+  open(): void {
+    this.do()
+      .then(() => this.stop())
+      .catch((error) => this.stop(error));
+  }
+
+  close(): void {
+    this.stop();
+  }
+
+  handleInput(data: string): void {
+    this.writeEmitter.fire(data === "\r" ? "\r\n" : data);
```

Comment:
Do we really need to support user inputs? From my understanding these tasks are output only?

Reply:
Since it is necessary for the user to enter `ctrl-c` (the user may want to manually end some background tasks), the input is echoed here. If the input is not echoed, the user may mistakenly believe that the window does not support any input. Echoing user input is the same as the behave of the shell terminal.

Reply:
I prefer no echo. For a non-background task, pressing `Ctrl-C` to terminate it is not supported. For a backgroud task, I think users know intuitively that he can press `Ctrl-C` to terminate it.

Reply:
> Echoing user input is the same as the behave of the shell terminal.

I think this is not the behavior of the shell, but the actual process/command. For example, `ngrok` does not echo and `nodemon` does echo.

If you want to implement something like Ctrl-C, I think the better way is just to send anything from input to the process/command's stdin. If it wants to echo it will echo, if it wants to handle Ctrl-C, they will quit on `'\x03'`, etc. Shell should not assume what the actual process/command want to do on keyboard input, Ctrl-C, etc.

Reply:
ok, I removed the echo from handle input.